src/python/botan3.py

```diff
@@ -631,14 +644,111 @@ def const_time_compare(x, y):
+        # that the underlying object lives at least as long as this context.
+        self.rng_ = rng
+        _DLL.botan_tpm2_ctx_enable_crypto_backend(self.handle_(), self.rng_.handle_())
+
+class TPM2Session(TPM2Object):
+    """Basic TPM 2.0 Session object, typically users will instantiate a derived class."""
+
+    def __init__(self, obj: c_void_p):
+        super().__init__(obj, _DLL.botan_tpm2_session_destroy)
+
+    @staticmethod
+    def session_bundle_(*args):
+        """Transforms a session bundle passed by the downstream user into a 3-tuple of session handles.
+        Users might pass a bare TPM2Session object or an iterable list of such objects."""
+        if len(args) == 1:
+            if isinstance(args[0], Iterable):
+                args = list(args[0])
+            elif args[0] is None:
+                args = []
+
+        if len(args) <= 3 and all(isinstance(s, TPM2Session) for s in args):
+            sessions = list(args)
+            while len(sessions) < 3:
+                sessions.append(None)
+            return (s.handle_() if isinstance(s, TPM2Session) else None for s in sessions)
+        else:
+            raise BotanException("session bundle arguments must be 0 to 3 TPM2Session objects")
+
+
+class TPM2UnauthenticatedSession(TPM2Session):
+    """Session object that is not bound to any authenication credential.
+    It provides basic parameter encryption between the application and the TPM."""
+    def __init__(self, ctx: TPM2Context):
+        obj = c_void_p(0)
+        _DLL.botan_tpm2_unauthenticated_session_init(byref(obj), ctx.handle_())
+        super().__init__(obj)
+
 #
 # RNG
 #
 class RandomNumberGenerator:
     # Can also use type "system"
-    def __init__(self, rng_type='system'):
+    def __init__(self, rng_type='system', **kwargs):
+        """Constructs a RandomNumberGenerator of type rng_type
+
+        Available RNG types are::
+
+        * 'system': Adapter to the operating system's RNG
+        * 'user':   Software-PRNG that is auto-seeded by the system RNG
+        * 'null':   Mock-RNG that fails if randomness is pulled from it
+        * 'hwrng':  Adapter to an available hardware RNG (platform dependent)
+        * 'tpm2':   Adapter to a TPM 2.0 RNG
+                    (needs additional named arguments tpm2_context= and, optionally, tpm2_sessions=)
+        """
         self.__obj = c_void_p(0)
-        _DLL.botan_rng_init(byref(self.__obj), _ctype_str(rng_type))
+        if rng_type == 'tpm2':
+            ctx = kwargs.get("tpm2_context", None)
+            if not ctx or not isinstance(ctx, TPM2Context):
+                raise BotanException("Cannot instantiate a TPM2-based RNG without a TPM2 context, pass tpm2_context= argument?")
+            sessions = TPM2Session.session_bundle_(kwargs.get("tpm2_sessions", None))
+            _DLL.botan_tpm2_rng_init(byref(self.__obj), ctx.handle_(), *sessions)
+        else:
+            _DLL.botan_rng_init(byref(self.__obj), _ctype_str(rng_type))
```

Comment:
Should check here that kwargs is otherwise empty

Reply:
Used `kwargs.pop()` and validated that `kwargs` is empty after reading all relevant arguments. Also added a test for that.